tests/test_salesforce_all_fields_non_custom.py

```diff
@@ -0,0 +1,48 @@
+"""
+Test that with only non-custom fields selected for a stream automatic fields and non custom fields  are still replicated
+"""
+
+from tap_tester.base_suite_tests.all_fields_test import AllFieldsTest
+from sfbase import SFBaseTest
+
+
+class SFNonCustomFieldsTest(AllFieldsTest, SFBaseTest):
+
+    salesforce_api = 'BULK'
+
+    @staticmethod
+    def name():
+        return "tt_sf_all_fields_non_custom"
+
+    def streams_to_test(self):
+        if self.partitioned_streams:
+            return self.partitioned_streams
+        return self.partition_streams(self.get_streams_with_data())
```

Comment:
This function is repeated 4 times and should be moved to `sfbase`

Reply:
All these tests are implement in the new tap-tester framework and the All_fields test needs this method to be present in the test as it is a abstract method in tap-tester all_fields test , if I understood it right. Do let me know if thats not the case.
https://github.com/stitchdata/tap-tester/blob/b280a171ce2ecc565d54f96bbef7cac9fe4a61ec/tap_tester/base_suite_tests/all_fields_test.py#L49

Reply:
I think if `SFBaseTest` implements it then `SFNonCustomFieldsTest` doesn't have to

Reply:
Let me try to write up a quick test